src/wp-includes/class-wp-image-editor-imagick.php

```diff
@@ -444,6 +444,39 @@ protected function thumbnail_image( $dst_w, $dst_h, $filter_name = 'FILTER_TRIAN
 		}
 
 		try {
+			$is_png                                      = false;
+			$is_indexed_png                              = false;
+			$is_indexed_png_with_alpha_channel           = false;
+			$is_indexed_png_with_true_alpha_transparency = false;
+
+			if ( 'image/png' === $this->mime_type ) {
+				$is_png = true;
+
+				if (
+					is_callable( array( $this->image, 'getImageProperty' ) )
+					&& '3' === $this->image->getImageProperty( 'png:IHDR.color-type-orig' )
+				) {
+					$is_indexed_png = true;
+
+					if (
+						is_callable( array( $this->image, 'getImageAlphaChannel' ) )
+						&& $this->image->getImageAlphaChannel()
+					) {
+						$is_indexed_png_with_alpha_channel = true;
+
+						if (
+							is_callable( array( $this->image, 'getImageChannelDepth' ) )
+							&& defined( 'Imagick::CHANNEL_ALPHA' )
```

Comment:
```suggestion
							&& defined( 'Imagick::CHANNEL_ALPHA' )
							&& 1 < $this->image->getImageChannelDepth( Imagick::CHANNEL_ALPHA )
```
something like this.